### PR TITLE
Add support for testing Azure storage backend in CI using Azurite.

### PR DESCRIPTION
**Pull request by ncabatoff**
This is exposing a problem with the Azurite List implementation and how it handles pagination, so this PR is on hold until they fix that.

### CODE REVIEW

File: physical/azure/azure.go [hunk 1]
```diff
@@ -84,17 +84,29 @@ func NewAzureBackend(conf map[string]string, logger log.Logger) (physical.Backen
...
 
 	var environment azure.Environment
+	var URL *url.URL
 	var err error
 
-	if environmentURL != "" {
-		environment, err = azure.EnvironmentFromURL(environmentURL)
-		if err != nil {
-			return nil, fmt.Errorf("failed to look up Azure environment descriptor for URL %q: %w", environmentURL, err)
+	testHost := conf["testHost"]
```
Comment (benashz) [inline physical/azure/azure.go:90]: I wonder if there is a possibility of `testHost` being configured outside the test? Not sure if this is a valid concern though.

**Review by benashz — APPROVED** [review]
One minor question/concern, otherwise LGTM!

File: physical/azure/azure.go (hunk 1 above)
Comment (ncabatoff) [reply physical/azure/azure.go:90]: Yes, this does modify prod code, so a Vault user could make use of this undocumented option.  I'm okay with that personally, but if you foresee a problem speak up.

Comment (benashz) [reply physical/azure/azure.go:90]: Perhaps we could rename `tesHost` to something that denotes that this configuration is not for prod. I would suggest something like `_testHost_` or  `_ciTestHost_`?

Comment (benashz) [reply physical/azure/azure.go:90]:
```suggestion
	// used by integration tests
	testHost := conf["_testHost_"]
```

Comment (ncabatoff) [reply physical/azure/azure.go:90]: Having the `test` prefix (and not documenting it) is good enough on that front for me.  There's plenty of examples of other such conf in our codebase, e.g. Raft's apply_delay: https://github.com/hashicorp/vault/blob/main/physical/raft/raft.go#L337

**Review by benashz — APPROVED** [review]

File: physical/azure/azure.go (hunk 1 above)
Comment (benashz) [reply physical/azure/azure.go:90]: Sounds good!

**Pull request merged**